bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/core/IBigtableDataClient.java

```diff
@@ -153,4 +155,12 @@
    */
   //TODO(rahulkql): Once it is adapted to v2.models, change the return type to ApiFuture.
   ListenableFuture<List<FlatRow>> readFlatRowsAsync(Query request);
+
+  /**
+   * Read {@link FlatRow} asynchronously, and pass them to a stream observer to be processed.
+   *
+   * @param request  a {@link Query} object.
+   * @param observer a {@link StreamObserver} object.
+   */
+  void readFlatRowsAsync(Query request, StreamObserver<FlatRow> observer);
```

Comment:
Whats the future plan of FlatRow vs model Row?

Reply:
I have no idea.  Let's discuss that offline.  For now, I'd rather continue as is.

Reply:
I created #2077 to track this.